openlibrary/plugins/upstream/borrow.py

```diff
@@ -160,10 +163,16 @@ def POST(self, key):
             user.update_loan_status()
             raise web.seeother(web.ctx.path)
         elif action == 'join-waitinglist':
+            waitinglist.WaitingLoan.get_cached_waiting_loan.memcache_delete(
```

Comment:
In the borrow endpoint we definitely do not want to use the cached version of this function. What we may want to do is **clear** the cache here.